src/main/resources/reference.conf

```diff
@@ -242,6 +255,19 @@ cassandra-snapshot-store {
   # Port of contact points in the cluster.
   # Will be ignored if the contact point list is defined by host:port pairs.
   port = 9042
+  
+  # An implementation of akka.persistence.cassandra.ContactPointsProvider
+  # can be used to lookup the contact points of the Cassandra cluster
+  # asynchronously instead of given them in the configuration. If this
+  # is specified the contact-points and port properties will be ignored.
+  # Define fully qualified class name of the implementation of the 
+  # akka.persistence.cassandra.ContactPointsProvider. It may optionally
+  # have a constructor with a ActorSystem parameter.
+  contact-points-provider = ""
+  
+  # The identifier that will be passed as parameter to the
+  # ContactPointsProvider.lookup method. 
+  contact-points-provider-id = ""
```

Comment:
> I don't think that is exactly the same. If C* is up and running I want to provide best effort for the initial persist requests, and don't randomly fail due to startup timing issues.

...but that's just the nature of distributed systems - there are no guarantees that Cassandra will be ready to receive your requests. There's a good chance that it will though, and even when using an actor selection, I'll bet that the majority of times will be just fine i.e. the selection will resolve.

Let it crash! :-)

> Nothing prevents you from delegating to your actor when implementing the lookup mehod. That would be one line using actor selection and ask.
 
Fair comment.